File: ruby-client/lib/rakam_client/api/realtime_api.rb
require "uri"

module RakamClient
  class RealtimeApi
    attr_accessor :api_client

    def initialize(api_client = ApiClient.default)
      @api_client = api_client
    end

    # Create report
    # 
    # @param real_time_report 
    # @param [Hash] opts the optional parameters
    # @return [JsonResponse]
    def realtime_create(real_time_report, opts = {})
      data, status_code, headers = realtime_create_with_http_info(real_time_report, opts)
      return data
    end

    # Create report
    # 
    # @param real_time_report 
    # @param [Hash] opts the optional parameters
    # @return [Array<(JsonResponse, Fixnum, Hash)>] JsonResponse data, response status code and response headers
    def realtime_create_with_http_info(real_time_report, opts = {})
      if @api_client.config.debugging
        @api_client.config.logger.debug "Calling API: RealtimeApi#realtime_create ..."
      end
      
      # verify the required parameter 'real_time_report' is set
      fail "Missing the required parameter 'real_time_report' when calling realtime_create" if real_time_report.nil?
      
      # resource path
      path = "/realtime/create".sub('{format}','json')

      # query parameters
      query_params = {}

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = ['application/json']
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = ['application/json']
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}

      # http body (model)
      post_body = @api_client.object_to_http_body(real_time_report)
      

      auth_names = ['master_key']
      data, status_code, headers = @api_client.call_api(:POST, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names,
        :return_type => 'JsonResponse')
      if @api_client.config.debugging
        @api_client.config.logger.debug "API called: RealtimeApi#realtime_create\nData: #{data.inspect}\nStatus code: #{status_code}\nHeaders: #{headers}"
      end
      return data, status_code, headers
    end

    # Delete report
    # 
    # @param [Hash] opts the optional parameters
    # @option opts [String] :project 
    # @option opts [String] :table_name 
    # @return [JsonResponse]
    def realtime_delete(opts = {})
      data, status_code, headers = realtime_delete_with_http_info(opts)
      return data
    end

    # Delete report
    # 
    # @param [Hash] opts the optional parameters
    # @option opts [String] :project 
    # @option opts [String] :table_name 
    # @return [Array<(JsonResponse, Fixnum, Hash)>] JsonResponse data, response status code and response headers
    def realtime_delete_with_http_info(opts = {})
      if @api_client.config.debugging
        @api_client.config.logger.debug "Calling API: RealtimeApi#realtime_delete ..."
      end
      
      # resource path
      path = "/realtime/delete".sub('{format}','json')

      # query parameters
      query_params = {}

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = ['application/json']
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = ['application/json']
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}
      form_params["project"] = opts[:'project'] if opts[:'project']
      form_params["table_name"] = opts[:'table_name'] if opts[:'table_name']

      # http body (model)
      post_body = nil
      

      auth_names = ['master_key']
      data, status_code, headers = @api_client.call_api(:POST, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names,
        :return_type => 'JsonResponse')
      if @api_client.config.debugging
        @api_client.config.logger.debug "API called: RealtimeApi#realtime_delete\nData: #{data.inspect}\nStatus code: #{status_code}\nHeaders: #{headers}"
      end
      return data, status_code, headers
    end

    # Get report
    # 
    # @param realtime_get 
    # @param [Hash] opts the optional parameters
    # @return [RealTimeQueryResult]
    def realtime_get(realtime_get, opts = {})
      data, status_code, headers = realtime_get_with_http_info(realtime_get, opts)
      return data
    end

    # Get report
    # 
    # @param realtime_get 
    # @param [Hash] opts the optional parameters
    # @return [Array<(RealTimeQueryResult, Fixnum, Hash)>] RealTimeQueryResult data, response status code and response headers
    def realtime_get_with_http_info(realtime_get, opts = {})
      if @api_client.config.debugging
        @api_client.config.logger.debug "Calling API: RealtimeApi#realtime_get ..."
      end
      
      # verify the required parameter 'realtime_get' is set
      fail "Missing the required parameter 'realtime_get' when calling realtime_get" if realtime_get.nil?
      
      # resource path
      path = "/realtime/get".sub('{format}','json')

      # query parameters
      query_params = {}

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = ['application/json']
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = ['application/json']
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}

      # http body (model)
      post_body = @api_client.object_to_http_body(realtime_get)
      

      auth_names = ['read_key']
      data, status_code, headers = @api_client.call_api(:POST, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names,
        :return_type => 'RealTimeQueryResult')
      if @api_client.config.debugging
        @api_client.config.logger.debug "API called: RealtimeApi#realtime_get\nData: #{data.inspect}\nStatus code: #{status_code}\nHeaders: #{headers}"
      end
      return data, status_code, headers
    end

    # List queries
    # 
    # @param [Hash] opts the optional parameters
    # @option opts [String] :project 
    # @return [Array<ContinuousQuery>]
    def realtime_list(opts = {})
      data, status_code, headers = realtime_list_with_http_info(opts)
      return data
    end

    # List queries
    # 
    # @param [Hash] opts the optional parameters
    # @option opts [String] :project 
    # @return [Array<(Array<ContinuousQuery>, Fixnum, Hash)>] Array<ContinuousQuery> data, response status code and response headers
    def realtime_list_with_http_info(opts = {})
      if @api_client.config.debugging
        @api_client.config.logger.debug "Calling API: RealtimeApi#realtime_list ..."
      end
      
      # resource path
      path = "/realtime/list".sub('{format}','json')

      # query parameters
      query_params = {}

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = ['application/json']
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = ['application/json']
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}
      form_params["project"] = opts[:'project'] if opts[:'project']

      # http body (model)
      post_body = nil
      

      auth_names = ['read_key']
      data, status_code, headers = @api_client.call_api(:POST, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names,
        :return_type => 'Array<ContinuousQuery>')
      if @api_client.config.debugging
        @api_client.config.logger.debug "API called: RealtimeApi#realtime_list\nData: #{data.inspect}\nStatus code: #{status_code}\nHeaders: #{headers}"
      end
      return data, status_code, headers
    end
  end
end

File: java-client/src/main/java/org/rakam/client/api/RealtimeApi.java
package org.rakam.client.api;

import com.sun.jersey.api.client.GenericType;

import org.rakam.ApiException;
import org.rakam.ApiClient;
import org.rakam.Configuration;
import org.rakam.Pair;

import org.rakam.client.model.RealTimeReport;
import org.rakam.client.model.JsonResponse;
import org.rakam.client.model.RealTimeQueryResult;
import org.rakam.client.model.RealtimeGet;
import org.rakam.client.model.ContinuousQuery;

import java.util.*;

@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JavaClientCodegen", date = "2016-05-06T02:48:07.215+03:00")
public class RealtimeApi {
  private ApiClient apiClient;

  public RealtimeApi() {
    this(Configuration.getDefaultApiClient());
  }

  public RealtimeApi(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  public ApiClient getApiClient() {
    return apiClient;
  }

  public void setApiClient(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  
  /**
   * Create report
   * 
   * @param realTimeReport 
   * @return JsonResponse
   */
  public JsonResponse realtimeCreate(RealTimeReport realTimeReport) throws ApiException {
    Object postBody = realTimeReport;
    
     // verify the required parameter 'realTimeReport' is set
     if (realTimeReport == null) {
        throw new ApiException(400, "Missing the required parameter 'realTimeReport' when calling realtimeCreate");
     }
     
    // create path and map variables
    String path = "/realtime/create".replaceAll("\\{format\\}","json");

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, Object> formParams = new HashMap<String, Object>();

    

    

    

    final String[] accepts = {
      "application/json"
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      "application/json"
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    String[] authNames = new String[] { "master_key" };

    
    GenericType<JsonResponse> returnType = new GenericType<JsonResponse>() {};
    return apiClient.invokeAPI(path, "POST", queryParams, postBody, headerParams, formParams, accept, contentType, authNames, returnType);
    
  }
  
  /**
   * Delete report
   * 
   * @param project 
   * @param tableName 
   * @return JsonResponse
   */
  public JsonResponse realtimeDelete(String project, String tableName) throws ApiException {
    Object postBody = null;
    
    // create path and map variables
    String path = "/realtime/delete".replaceAll("\\{format\\}","json");

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, Object> formParams = new HashMap<String, Object>();

    

    

    if (project != null)
      formParams.put("project", project);
    if (tableName != null)
      formParams.put("table_name", tableName);
    

    final String[] accepts = {
      "application/json"
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      "application/json"
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    String[] authNames = new String[] { "master_key" };

    
    GenericType<JsonResponse> returnType = new GenericType<JsonResponse>() {};
    return apiClient.invokeAPI(path, "POST", queryParams, postBody, headerParams, formParams, accept, contentType, authNames, returnType);
    
  }
  
  /**
   * Get report
   * 
   * @param realtimeGet 
   * @return RealTimeQueryResult
   */
  public RealTimeQueryResult realtimeGet(RealtimeGet realtimeGet) throws ApiException {
    Object postBody = realtimeGet;
    
     // verify the required parameter 'realtimeGet' is set
     if (realtimeGet == null) {
        throw new ApiException(400, "Missing the required parameter 'realtimeGet' when calling realtimeGet");
     }
     
    // create path and map variables
    String path = "/realtime/get".replaceAll("\\{format\\}","json");

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, Object> formParams = new HashMap<String, Object>();

    

    

    

    final String[] accepts = {
      "application/json"
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      "application/json"
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    String[] authNames = new String[] { "read_key" };

    
    GenericType<RealTimeQueryResult> returnType = new GenericType<RealTimeQueryResult>() {};
    return apiClient.invokeAPI(path, "POST", queryParams, postBody, headerParams, formParams, accept, contentType, authNames, returnType);
    
  }
  
  /**
   * List queries
   * 
   * @param project 
   * @return List<ContinuousQuery>
   */
  public List<ContinuousQuery> realtimeList(String project) throws ApiException {
    Object postBody = null;
    
    // create path and map variables
    String path = "/realtime/list".replaceAll("\\{format\\}","json");

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, Object> formParams = new HashMap<String, Object>();

    

    

    if (project != null)
      formParams.put("project", project);
    

    final String[] accepts = {
      "application/json"
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      "application/json"
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    String[] authNames = new String[] { "read_key" };

    
    GenericType<List<ContinuousQuery>> returnType = new GenericType<List<ContinuousQuery>>() {};
    return apiClient.invokeAPI(path, "POST", queryParams, postBody, headerParams, formParams, accept, contentType, authNames, returnType);
    
  }
  
}
